src/main/scala/ophan/google/index/checker/GoogleSearchService.scala

```diff
@@ -1,56 +1,90 @@
 package ophan.google.index.checker
 
-import com.google.api.client.http.HttpRequest
-import com.google.api.client.http.javanet.NetHttpTransport
-import com.google.api.client.json.gson.GsonFactory
-import com.google.api.services.customsearch.v1.CustomSearchAPI
-import com.google.api.services.customsearch.v1.model.{Result, Search}
 import ophan.google.index.checker.GoogleSearchService.resultMatches
-import ophan.google.index.checker.model.{CheckReport, ContentSummary}
+import upickle.default.*
 
 import java.net.URI
+import java.net.http.{HttpClient, HttpRequest}
 import java.time.Instant
 import scala.concurrent.{ExecutionContext, Future, blocking}
-import scala.jdk.CollectionConverters._
 import scala.util.Try
+import ophan.google.index.checker.model.{CheckReport, ContentSummary}
+
+case class DerivedStructData(link: String)
+
+object DerivedStructData {
+  implicit val rw: ReadWriter[DerivedStructData] = macroRW
+}
+
+case class Document(derivedStructData: DerivedStructData)
+
+object Document {
+  implicit val rw: ReadWriter[Document] = macroRW
+}
+
+case class SearchResult(document: Document)
+
+object SearchResult {
+  implicit val rw: ReadWriter[SearchResult] = macroRW
+}
+
+case class SearchResponse(results: List[SearchResult] = List.empty)
+
+object SearchResponse {
+  implicit val rw: ReadWriter[SearchResponse] = macroRW
+}
 
 class GoogleSearchService(
-  apiKey: String
-)(implicit
-  ec: ExecutionContext
-) {
-  val search =
-    new CustomSearchAPI.Builder(
-      new NetHttpTransport,
-      new GsonFactory,
-      (request: HttpRequest) => {
-        request.getHeaders.setUserAgent("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36")
+                           projectId: String,
+                           location: String,
+                           dataStoreId: String,
+                           apiKey: String
+                         )(implicit ec: ExecutionContext) {
+
+  private val baseUrl = s"https://discoveryengine.googleapis.com/v1/projects/$projectId/locations/$location/dataStores/$dataStoreId/servingConfigs/default_config:searchLite"
+
+  private val httpClient = HttpClient.newHttpClient()
+
+  def contentAvailabilityInGoogleIndex(content: ContentSummary): Future[CheckReport] = Future {
+    blocking {
+      def performSearch(searchTerm: String): Boolean = {
+        val requestBody = ujson.Obj(
+          "query" -> searchTerm,
+          "pageSize" -> 10
+        )
 
+        val request = HttpRequest.newBuilder()
+          .uri(URI.create(s"$baseUrl?key=$apiKey"))
+          .header("Content-Type", "application/json")
+          .header("User-Agent", "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36")
```

Comment:
Why does this need to be set?

Reply:
Good question, just copied it, ill have a look

Reply:
This shouldn't need to be set, I will double check and test